packages/grid/x-data-grid/src/components/columnHeaders/GridColumnGroupHeader.tsx

```diff
@@ -0,0 +1,157 @@
+  const { classes, headerAlign, isDragging, showRightBorder, showColumnBorder, groupId } =
+    ownerState;
+
+  const slots = {
+    root: [
+      'columnHeader',
+      headerAlign === 'left' && 'columnHeader--alignLeft',
+      headerAlign === 'center' && 'columnHeader--alignCenter',
+      headerAlign === 'right' && 'columnHeader--alignRight',
+      isDragging && 'columnHeader--moving',
+      showRightBorder && 'withBorder',
+      showColumnBorder && 'columnHeader--showColumnBorder',
+      groupId === null ? 'columnHeader--emptyGroup' : 'columnHeader--filledGroup',
+    ],
+    draggableContainer: ['columnHeaderDraggableContainer'],
+    titleContainer: ['columnHeaderTitleContainer'],
+    titleContainerContent: ['columnHeaderTitleContainerContent'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+function GridColumnGroupHeader(props: GridColumnGroupHeaderProps) {
+  const {
+    groupId,
+    width,
+    depth,
+    maxDepth,
+    fields,
+    height,
+    colIndex,
+    isLastColumn,
+    extendRowFullWidth,
+  } = props;
+
+  const rootProps = useGridRootProps();
+
+  const apiRef = useGridApiContext();
+  const columnGroupsLookup = useGridSelector(apiRef, gridColumnGroupsLookupSelector);
+  const { hasScrollX, hasScrollY } = apiRef.current.getRootDimensions() ?? {
+    hasScrollX: false,
+    hasScrollY: false,
+  };
+
+  const group: Partial<GridColumnGroup> = groupId ? columnGroupsLookup[groupId] : {};
+
+  const { headerName = groupId ?? '', description = '', headerAlign = undefined } = group;
+
+  let headerComponent: React.ReactNode;
+
+  const render = groupId && columnGroupsLookup[groupId]?.renderHeaderGroup;
+  const renderParams = {
+    groupId,
+    headerName,
+    description,
+    depth,
+    maxDepth,
+    fields,
+    colIndex,
+    isLastColumn,
+  };
+  if (groupId && render) {
+    headerComponent = render(renderParams);
+  }
+
+  const removeLastBorderRight = isLastColumn && hasScrollX && !hasScrollY;
+  const showRightBorder = !isLastColumn
+    ? rootProps.showColumnRightBorder
+    : !removeLastBorderRight && !extendRowFullWidth;
+
+  const showColumnBorder = rootProps.showColumnRightBorder;
+
+  const ownerState = {
+    ...props,
+    classes: rootProps.classes,
+    showRightBorder,
+    showColumnBorder,
+    headerAlign,
+    depth,
+    isDragging: false,
+  };
+
+  const label = headerName ?? groupId;
+
+  const id = useId();
+  const elementId = groupId === null ? `empty-group-cell-${id}` : groupId;
+  const classes = useUtilityClasses(ownerState);
+
+  const headerClassName =
+    typeof group.headerClassName === 'function'
+      ? group.headerClassName(renderParams)
+      : group.headerClassName;
+
+  return (
+    <GridGenericColumnHeaderItem
+      classes={classes}
+      columnMenuOpen={false}
+      colIndex={colIndex}
+      height={height}
+      isResizing={false}
+      sortDirection={null}
+      hasFocus={false}
+      tabIndex={0}
```

Comment:
The column group headers don't have keyboard navigation with the arrow keys, but with `tabIndex=0` we can navigate with <kbd>Tab</kbd> between them. This is breaks the idea of only-one-focusable-element-per-grid from https://www.w3.org/WAI/ARIA/apg/patterns/grid/. I think they should be non-interactive.

```suggestion
      tabIndex={-1}
```